src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -63,23 +64,21 @@ class ErgoStateContext(val lastHeaders: Seq[Header],
 
   override type M = ErgoStateContext
 
-  def sigmaPreHeader: special.sigma.PreHeader = PreHeader.toSigma(lastHeaders.head)
+  override def sigmaPreHeader: special.sigma.PreHeader =
+    PreHeader.toSigma(lastHeaders.headOption.getOrElse(PreHeader.fake))
```

Comment:
@aslesarenko I agree that explicit NPE is better. But if we make `ErgoLikeContext.preHeader` to be `Option[PreHeader]` but left it non-optional in script `CONTEXT.preHeader` we would crash in runtime, right? Why is it better than do a hard unwrap and crash here?

Reply:
It is not a problem if the script crashes at runtime, this means `false`.
And I don't see how the code "do a hard unwrap and crash here".

Reply:
You're right, we don't have to crash, we return `false`. But making `ErgoLikeContext.preHeader` optional would have bigger implications since we take `HEIGHT` and `minerPk` from it. And since it cannot be null by design I believe we should express this in types. Otherwise, it should be optional in ergo as well.